mmcv/fileio/file_client.py

```diff
@@ -222,16 +222,35 @@ def __init__(self, backend='disk', **kwargs):
         self.client = self._backends[backend](**kwargs)
 
     @classmethod
-    def register_backend(cls, name, backend):
+    def _register_backend(cls, name, backend, force=False):
+        if not isinstance(name, str):
+            raise TypeError('the backend name should be a string, '
+                            f'but got {type(name)}')
         if not inspect.isclass(backend):
             raise TypeError(
                 f'backend should be a class but got {type(backend)}')
         if not issubclass(backend, BaseStorageBackend):
             raise TypeError(
                 f'backend {backend} is not a subclass of BaseStorageBackend')
+        if not force and name in cls._backends:
```

Comment:
assert hasattr(cls, '_backends')

Reply:
This is always true, for both FileClient and subclasses.